Firestore/src/Query.php

```diff
@@ -1014,4 +978,144 @@ private function basePath()
+                    'field' => [
+                        'fieldPath' => $escapedPathString
+                    ],
+                    'op' => $unaryOperator
+                ]
+            ];
+        } else {
+            $encodedValue = $operator === FieldFilterOperator::IN
+                ? $this->valueMapper->encodeMultiValue((array)$value)
+                : $this->valueMapper->encodeValue($value);
+
+            $filter = [
+                'fieldFilter' => [
+                    'field' => [
+                        'fieldPath' => $escapedPathString,
+                    ],
+                    'op' => $operator,
+                    'value' => $encodedValue
+                ]
+            ];
+        }
+        
+        return $filter;
+    }
+
+    /**
+     * Find and set the the filter type to either Field or Unary.
+     *
+     * @param array $filter A field/unary filter array.
+     * @return array A non composite Filter array.
+     * @throws \InvalidArgumentException If an invalid filter is encountered.
+     */
+    private function findAndSetNonCompositeFilter($filter)
+    {
+        $nonCompositeFilter = $this->createNonCompositeFilter(
+            $filter['fieldFilter']['field'],
+            $filter['fieldFilter']['op'],
+            $filter['fieldFilter']['value']
+        );
+
+        return $nonCompositeFilter;
+    }
+
+    /**
+     * Create a composite filter from a filter array.
+     *
+     * @param array $filters A non sanitized composite filter array.
+     * @return array A sanitized composite Filter array.
+     * @throws \InvalidArgumentException If an invalid filter is encountered.
+     */
+    private function createCompositeFilter($filters)
+    {
+        foreach ($filters as $key => $filter) {
+            if ($key === 'fieldFilter') {
+                $nonCompositeFilter = $this->findAndSetNonCompositeFilter($filters);
+                if (isset($nonCompositeFilter['unaryFilter'])) {
+                    unset($filters['fieldFilter']);
+                    $filters['unaryFilter'] = $nonCompositeFilter['unaryFilter'];
+                } else {
+                    $filters['fieldFilter'] = $nonCompositeFilter['fieldFilter'];
+                }
+            } elseif (isset($filter['fieldFilter'])) {
+                $filters[$key] = $this->findAndSetNonCompositeFilter($filter);
+            } elseif ($key === 'compositeFilter' && isset($filter['filters'])) {
+                $filters[$key]['filters'] = $this->createCompositeFilter($filter['filters']);
+            } elseif (isset($filter['compositeFilter']) &&
+            isset($filter['compositeFilter']['filters'])) {
+                $filters[$key]['compositeFilter']['filters'] =
+                    $this->createCompositeFilter($filter['compositeFilter']['filters']);
+            } else {
+                throw new \InvalidArgumentException(
+                    'The filter is not set correctly.'
+                );
+            }
```

Comment:
Use implicit else

```suggestion
            }
            throw new \InvalidArgumentException(
                'The filter is not set correctly.'
            );
```

Reply:
Since we are not doing the `return` from the above `if`/`elseif`s, if we remove the last else, it will always come to that throw exception no?

Reply:
OMG, thanks for not taking this suggestion. I didn't realize that this function returns a value. My bad!